src/Pages/Signup.js

```diff
@@ -0,0 +1,116 @@
+import React,{useReducer,useState,useEffect} from 'react'
+import axios from 'axios'
+import {Link,useNavigate} from 'react-router-dom'
+
+export function Signup() {
+    const navigate = useNavigate()
+
+    const [textInputState,textInputDispatcher] = useReducer(inputDispatcher,{username:"",password:"",rePassword:"",answer:""})
+    
+    const [visibilityErr,setErrVisibility] = useState("hidden")
+    const [Alertclass,setAlertClass] = useState("alert")
+    const [faClass,setFaClass] = useState("fa-check-circle-o")
+    const [content,setContent] = useState("")
+
+    const [signupContent,setSignupContent] = useState("Sign up")
+    async function userSignup(){
```

Comment:
As this function is lengthy and occupies a lot of space, it can be extracted to a different file, for better readability.